.github/workflows/lint.yml

```diff
@@ -23,15 +23,19 @@ jobs:
     - name: Setup Python
       uses: actions/setup-python@v2
       with:
-        python-version: 3.8 
+        python-version: 3.8
 
     - name: Install dependencies
-      run: pip install -U flake8 wheel setuptools
+      run: pip install -U flake8 wheel setuptools yapf==0.31.0
+
+    - name: Test that yapf has been applied
+      # If this check fails for your PR, run `yapf -rip .`
+      run: yapf --recursive --parallel --diff .
```

Comment:
Isn't `flake8` enough?

Reply:
I think having PRs go through `yapf` is a good idea, as it ensures formatting consistency in the long run. 

Also, if we merge couple of non-yapf-formatted PRs, the next time someone does yapf-format their branch before opening a PR, they are going to have to deal with formatting changes stemming from those preceding PRs as well (or ensure that only their diff is yapf-formatted).

Reply:
flake8 is not enough. yapf will still want to rewrap long lines. A PR wrapped to 80 characters will pass flake8 but fail that yapf check. And as Rok says, if a non yapf formatted PR gets merged then we'll be faced with the choice between having to do another *run yapf* commit or require every future contributor to manually filter yapf's changes to only the code which they wrote so as to dodge the non yapf formatted portion.

Reply:
Unless we opt to never using yapf again and only using flake8 in future?

Reply:
yeah ok makes sense, let's stick with `yapf` for a while - we can always drop it later for `flake8` if needs be.